packages/app/src/vis-packs/nexus/containers/NxComplexImageContainer.tsx

```diff
@@ -34,13 +35,13 @@ function NxComplexImageContainer(props: VisContainerProps) {
       />
       <VisBoundary resetKey={dimMapping}>
         <NxValuesFetcher
-          nxData={nxData}
+          nxData={nxData as NxData<ComplexType>}
```

Comment:
Unrelated improvement. This type cast is a bit nicer than the one below. The problem is that `assertComplexType` above asserts the type of the dataset and not of the `nxData` object... I was about to add a guard called `assertNxDataComplexType` or something, but it felt too much, especially sicne we'd need another guard checking for the other type in `NxImageContainer`.

Reply:
>  I was about to add a guard called `assertNxDataComplexType` or something, but it felt too much, especially sicne we'd need another guard checking for the other type in `NxImageContainer`.

I don't see why it would feel too much. I think it would make sense to assert the type of `NxData` as everything would follow, no ? (i.e it would allow to infer the type of `signalDataset` which would in turn would allow to infer the type of `signal` )